src/docs/development/ui/navigation/deep-linking.md

```diff
@@ -0,0 +1,126 @@
+matching the configured scheme will be handled by the `Navigator`. By following
+these steps, your can launch and display routes via named routes (either with
+the [`routes`][routes] parameter or [`onGenerateRoute`][onGenerateRoute]), or by
+using the [Router][] widget.
+
+If you're running the app in a web browser,  there's no additional setup
+required. Route paths are handled in the same way as an iOS or Android deep
+link. Web apps currently read the path from the url fragment by default
+(`/#/path/to/app/screen`)
+
+To follow along, create a new Flutter project with [the Router widget
+sample][router-sample] in flutter/samples.
+
+## Enable deep linking on Android
+
+Add a metadata tag and [intent filter][] to `AndroidManifest.xml`
+inside the `<activity> `tag with the `".MainActivity"` name:
+
+```
+<!-- Deep linking -->
+<meta-data android:name="flutter_deeplinking_enabled" android:value="true" />
+<intent-filter android:autoVerify="true">
+    <action android:name="android.intent.action.VIEW" />
+    <category android:name="android.intent.category.DEFAULT" />
+    <category android:name="android.intent.category.BROWSABLE" />
+    <data android:scheme="http" android:host="flutterbooksample.com" />
+    <data android:scheme="https" />
+</intent-filter>
+```
+
+A full restart is required to apply these changes.
+
+## Test on Android emulator
+To test with an Android emulator, the `adb` command can be given an intent. The
+host name should match the name defined in `AndroidManifest.xml`:
+
+
+```
+adb shell am start -a android.intent.action.VIEW \
+    -c android.intent.category.BROWSABLE \
+    -d "http://flutterbooksample.com/book/1"
+```
+
+
+See the [Verify Android App Links][verify-android-links] documentation for more
+details.
+
+## Enable deep linking on iOS
+Add two new keys to `Info.plist` in the ios/Runner directory:
+
+```
+<key>FlutterDeepLinkingEnabled</key>
+<true/>
+<key>CFBundleURLTypes</key>
+<array>
+    <dict>
+    <key>CFBundleTypeRole</key>
+    <string>Editor</string>
+    <key>CFBundleURLName</key>
+    <string>flutterbooksample.com</string>
+    <key>CFBundleURLSchemes</key>
+    <array>
+    <string>customscheme</string>
+    </array>
+    </dict>
+</array>
+```
+
+The `CFBundleURLName` is a unique URL used to distinguish your app from others
+that use the same scheme. The scheme (`customscheme://`)  can also be unique to
+help avoid conflicts.
+
+A full restart is required to apply these changes.
+
+## Test on iOS simulator
+The `xcrun` command can be used to test on the iOS Simulator:
+
+```
+xcrun simctl openurl booted customscheme://flutterbooksample.com/book/1 
+```
+
+## Migrating from plugin-based deep linking
+
+If you have written a plugin to handle deep links, as described in ["Deep Links
+and Flutter applications" on Medium][plugin-linking], it will continue to work
+until you opt-in to this behavior by adding `FlutterDeepLinkingEnabled` to
```

Comment:
Adding the property is optional for iOS to enable deeplink. Flutter Deep Link is handled by default for iOS.

or the behavior need to be fixed, as it defers from Android.

Reply:
https://github.com/flutter/engine/pull/22356 should disable it unless `FlutterDeepLinkingEnabled` is provided in `Info.plist`.

Reply:
@johnpryan If you look into the code here: https://github.com/flutter/engine/blob/6ba12fa2300e027916a928e2a617956f6a9763e8/shell/platform/darwin/ios/framework/Source/FlutterAppDelegate.mm#L135-L177

The deeplink is disabled if `FlutterDeeplinkEnabled` property if false.Otherwise it's handled anyway.

Reply:
> The deeplink is disabled if FlutterDeeplinkingEnabled property is false. Otherwise it's handled anyway.

`IsDeepLinkingEnabled` returns NO if `FlutterDeepLinkingEnabled`. So this feature is opt-in:

```
static BOOL IsDeepLinkingEnabled(NSDictionary* infoDictionary) {
  NSNumber* isEnabled = [infoDictionary objectForKey:@"FlutterDeepLinkingEnabled"];
  if (isEnabled) {
    return [isEnabled boolValue];
  } else {
    return NO;
  }
}
```

Reply:
Shams, this section applies to both Android and iOS.

Reply:
@johnpryan Please look for the usage of the method you added.

```
if( not IsDeepLinkingEnabled)
    return NO
else 
    handle deeplink
```

Reply:
SG